src/api/route-services/pipeline-response.js

```diff
@@ -78,9 +80,54 @@ const pipelineResponse = async (io, message) => {
     ]);
   }
 
-
   const statusRes = await pipelineStatus(experimentId);
 
+  if (taskName === 'configureEmbedding') {
+    const now = new Date();
+    const addTimeoutSeconds = 60;
+    const timeout = new Date(now.getTime() + addTimeoutSeconds * 1000);
+
+    // Run work request for embedding
+    const embeddingWorkRequest = {
+      experimentId,
+      timeout,
+      uuid: uuidv4(),
+      body: {
+        name: 'GetEmbedding',
+        type: output.config.embeddingSettings.method,
+        config: output.config.embeddingSettings.methodSettings[
+          output.config.embeddingSettings.method
+        ],
+      },
+      PipelineRunETag: statusRes.pipeline.startDate,
+    };
```

Comment:
I am slightly concerned that this piece of code as it stands is very coupled with the schema of the compute embedding task and if we wanted to change it in the future, which makes it very fragile and easy to break. Is there a way to abstract away the specific parameter setup and have it more generic here?

Reply:
This is refactored into `hookRunner` and associated functions under `src/utils/hooks`